File: src/main/java/red/tetracube/iot/smartigloohub/data/entities/TrustedBrokerClient.java
package red.tetracube.iot.smartigloohub.data.entities;

import red.tetracube.iot.smartigloohub.data.enumerations.ClientType;

import java.util.Collections;
import java.util.List;
import java.util.UUID;
import javax.persistence.*;

@Entity
@Table(schema = "igloo_hub", name = "trusted_broker_clients")
public class TrustedBrokerClient {

    @Id
    private UUID id;

    @Column(name = "client_id", nullable = false, unique = true)
    private String clientId;

    @Column(name = "username", nullable = false, unique = true)
    private String username;

    @Column(name = "password", nullable = false)
    private String password;

    @Enumerated(value = EnumType.STRING)
    @Column(name = "client_type", nullable = false)
    private ClientType clientType;

    @OneToMany(fetch = FetchType.LAZY, targetEntity = Switcher.class, mappedBy = "trustedBrokerClient")
    private List<Switcher> environments = Collections.emptyList();

    public String getPassword() {
        return password;
    }
}

File: src/main/kotlin/red/tetracube/iot/smartigloohub/data/entities/TrustedBrokerClient.kt
package red.tetracube.iot.smartigloohub.data.entities

import red.tetracube.iot.smartigloohub.data.enumerations.ClientType
import java.util.UUID
import javax.persistence.*

@Entity
@Table(schema = "igloo_hub", name = "trusted_broker_clients")
data class TrustedBrokerClient(
    @Id
    val id: UUID? = null,

    @Column(name = "client_id", nullable = false, unique = true)
    val clientId: String? = null,

    @Column(name = "username", nullable = false, unique = true)
    val username: String? = null,

    @Column(name = "password", nullable = false)
    val password: String? = null,

    @Enumerated(value = EnumType.STRING)
    @Column(name = "client_type", nullable = false)
    val clientType: ClientType? = null,

    @OneToMany(fetch = FetchType.LAZY, targetEntity = Switcher::class, mappedBy = "trustedBrokerClient")
    val environments: List<Switcher> = emptyList()
)
